### PR TITLE
Conditional skip of stopping resources in interactive mode

### PR DESCRIPTION
**Pull request by M6AI**
## Bug / Requirement Description
If an environment is started then stopped in interactive mode, then the abort would once again initiate the full stop for the environment, resulting in duplicated work and a UID clash for before and after stop hooks.

## Solution description
Added a simple conditional skip of this step based on environment status.

## Checklist:
- [ ] Test
- [ ] Example (both test_plan.py and .rst)
- [ ] Documentation (API)
- [ ] News fragment present for release notes
- [ ] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: testplan/runnable/interactive/base.py [hunk 1]
```diff
@@ -513,6 +513,9 @@ def stop_test_resources(self, test_uid, await_results=True):
         if not await_results:
             return self._run_async(self.stop_test_resources, test_uid)
 
+        if self.report[test_uid].env_status == entity.ResourceStatus.STOPPED:
+            return
```
Comment (M6AI) [inline testplan/runnable/interactive/base.py:517]: This is a performance boost in general, if something is stopped we simply skip the entire process. Not sure if there is a way to trigger another stop of the environment resources than usual (in which case the button status protects us), but just to be sure.

Comment (Pyifan) [reply testplan/runnable/interactive/base.py:517]: It's better that we check env_status not in [STOPPING, STOPPED]

File: testplan/testing/multitest/base.py [hunk 2]
```diff
@@ -709,6 +709,8 @@ def stop_test_resources(self):
         interactive mode where a test environment may be stopped on-demand.
         Handles running before/after_stop callables if required.
         """
+        if self.report.env_status == entity.ResourceStatus.STOPPED:
```
Comment (Pyifan) [inline testplan/testing/multitest/base.py:712]: it seems report.env_status is not something that we handle at Test or Multitest level. It is currently handled by TestRunnerIHandler. Or maybe we can move this state transition logic to Test. Anyway, it's better we set and check in the same class.

**Review by Pyifan — CHANGES_REQUESTED** [review]

File: testplan/testing/multitest/base.py (hunk 2 above)
Comment (M6AI) [reply testplan/testing/multitest/base.py]: For now I addressed the other two comments. I think the Environment itself should know about the overall status of its drivers and derive from that. It does not seem to me a Test or MultiTest level thing, only if it derives from the actual environment.

**Pull request closed without merging**